lib/mongoid/association/referenced/has_many/enumerable.rb

```diff
@@ -219,11 +219,18 @@ def empty?
             # Does the enumerable have any? Will determine if there exists at least
             # one element in the enumerable, whether or not it is _loaded.
             #
+            # Note that in Enumerable's any? method if both a pattern and a block are
+            # passed in, only the pattern is used. I implemented the same functionality
+            # here.
+            #
             # @example Doess the enumerable have any?
             #   enumerable.any?
             #
             # @return [ true, false ] If the enumerable has any elements.
-            def any?
+            def any?(pattern=nil, &block)
+              return super(pattern) if pattern
+              return super(&block) if block_given?
```

Comment:
Both of these calls should forward both arguments to super, i.e. I believe they can be simplified to 

```
return super if pattern || block_given?
```

Reply:
This doesn't seem to work, actually.

Reply:
When you say it doesn't work, what behavior are you observing?

Reply:
The block is not being passed along in the super call

Reply:
Thanks for explaining!

Reply:
You still need to pass both arguments in both cases, you could try `super(pattern, &block)` if needed.